### DISCUSSION TITLE
Exposing Port 8888 on tf-torch-conda version

### DISCUSSION THREAD

**Discussion by mpasha10** [matifali/dockerdl#27] [Ideas]
Hi, 

I am very much interested in your docker.  Would it be possible to expose 8888 and run Jupyter Lab by default when container is loaded?  currently, this cannot be done and when I start it with in the container, the port is not exposed outside!

Many thanks

**Comment by matifali** [comment]
What do you get when you run?
```shell
docker run --gpus all --rm -it -h dockerdl -p 8888:8888 matifali/dockerdl:tf-torch-conda jupyter lab --no-browser --port 8888 --ServerApp.token='' --ip='*'
```

---

I intentionally did not add jupyter-lab as the `CMD`. You can build a custom image using `matifali/dockerdl` as base and update the `CMD`.

Something like this should work.

```Dockerfile
FROM matifali/dockerdl:tf-torch-conda
CMD ["jupyter", "lab", "--no-browser", "--port", "8888", "--ServerApp.token=''", "--ip='*'"]
```
Then build the image and run it with
`docker run -p 8888:8888 username/iamge:tag`

**Reply by mpasha10** [reply]
both returns the following:
`/opt/nvidia/nvidia_entrypoint.sh: line 67: exec: jupyter: not found`

**Reply by matifali** [reply]
Interesting. 
Also try to update the `ENTRYPOINT`, as we are not updating it in Dockerfile and it defaults to the base image's entrypoint.

```Dockerfile
FROM matifali/dockerdl:tf-torch-conda
ENTRYPOINT ["jupyter", "lab", "--no-browser", "--port=8888", "--ip=0.0.0.0", "--allow-root"]
```
